cmake/Modules/SetGPUArchs.cmake

```diff
@@ -0,0 +1,61 @@
+# =============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# Build the list of supported architectures
+
+# Check for embedded vs workstation architectures
```

Comment:
RMM doesn't have many (any?) architecture-specific features. So it doesn't make a lot of sense to maintain a manually created list of supported architectures. It would be nice if we didn't have a file we have to update every time a new architecture comes out...

Reply:
Because RMM is header only this is only used for building the Tests and Benchmarks. We could change the behavior to only build for the current architecture, but this would break the semantics that the rest of the RAPIDS libraries follow and what we've historically done so I kept it the same.

Any thoughts / preferences?

Reply:
I don't like the idea of a list of architectures that we have to add each new architecture released by NVIDIA. I don't like it *anywhere* in RAPIDS.  Can we just specify a minimum arch version, and then explicitly disable any arches that we don't support? This way, we only have to add a new arch to the file if it is a special case that we do not support.

Reply:
CMake doesn't maintain a list of supported architectures for a given CUDA version as far as I know, so unfortunately we need to list the archs we build for somewhere.

Ideally this file will be put into a separate `rapids-core` repo in the future and reused across all of the RAPIDS libraries in the future but we're not quite there yet.

Reply:
That's not my point. My point is that the list should be a disallow list, rather than an allow list. Allow lists require us to edit them every time a new arch appears. disallow lists only require editing if we need to disable a new architecture.

Reply:
Got it! Apologies, wasn't clicking in my head. Will make it remove archs from the list as opposed to adding them.

Reply:
@harrism please take a look at latest changes 😄

Reply:
So there's still a list we have to maintain. Each time NVIDIA releases a new architecture, we have to edit this file. :(

Reply:
I guess I don't understand the purpose of this functionality.

Reply:
Discussed offline, I'm informed and on board now.